fv/datastore/datastore_describe.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2018 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package datastore
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+)
+
+type DatastoreType int
+
+const (
+	DatastoreEtcdV3 DatastoreType = 1 << iota
+	DatastoreK8s
+
+	DatastoreAll = DatastoreEtcdV3 | DatastoreK8s
+)
+
+// DatastoreDescribe is a replacement for ginkgo.Describe which invokes Describe
+// multiple times for one or more different datastore drivers - passing in the
+// Calico API configuration as a parameter to the test function.  This allows
+// easy construction of end-to-end tests covering multiple different datastore
+// drivers.
+//
+// The *datastores* parameter is a bit-wise OR of the required datastore drivers
+// that will be tested.
+func DatastoreDescribe(description string, datastores DatastoreType, body func(getInfra func() (DatastoreInfra, error))) bool {
+	if datastores&DatastoreEtcdV3 != 0 {
+		Describe(fmt.Sprintf("%s (etcdv3 backend)", description),
+			func() {
+				body(CreateEtcdDatastoreInfra)
```

Comment:
Guess these infra funcs aren't used outside this package, so should begin with lowercase.